### ISSUE TITLE
[XCode 11 | X.iOS 12.99.3.5] Device builds fail (MT3001 AOT Failed, error code 139)

### ISSUE THREAD

**Issue by projectgoav** [xamarin/xamarin-macios#6839]
### Steps to Reproduce

1. Open a Xamarin.iOS Solution
2. Select deployment to a device
3. Start debugging

### Expected Behavior

Build will complete and debugging session will start 

### Actual Behavior

Build error:

```
/Users/ewan/Documents/<path...to...app>/MTOUCH: Error MT3001: Could not AOT the assembly '/Users/ewan/Documents/<path...to....app>/obj/iPhone/Debug/device-builds/iphone8.1-12.4/mtouch-cache/3-Build/<app>.exe' (MT3001) (<app>)
```

### Environment

```
=== Visual Studio Community 2019 for Mac ===

Version 8.2.4 (build 45)
Installation UUID: 6e3bfb91-18aa-421f-9341-cdec6ad3af19
	GTK+ 2.24.23 (Raleigh theme)
	Xamarin.Mac 5.14.0.85 (d16-2-fix-mac-updates / 86973979)

	Package version: 600000327

=== Mono Framework MDK ===

Runtime:
	Mono 6.0.0.327 (2019-02/f8ea05bddcb) (64-bit)
	Package version: 600000327

=== NuGet ===

Version: 5.1.0.6013

=== .NET Core ===

Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	2.1.12
	2.1.11
	2.1.9
	2.1.8
	2.1.7
	2.1.2
	2.1.1
	2.0.5
	2.0.0
SDK: /usr/local/share/dotnet/sdk/2.1.701/Sdks
SDK Versions:
	2.1.701
	2.1.700
	2.1.505
	2.1.504
	2.1.503
	2.1.302
	2.1.301
	2.1.4
	2.0.0
MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/6.0.0/lib/mono/msbuild/Current/bin/Sdks

=== Xamarin.Profiler ===

Version: 1.6.11
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

=== Updater ===

Version: 11

=== Apple Developer Tools ===

Xcode 11.0 (14921.5)
Build 11M392q

=== Xamarin.Mac ===

Version: 5.99.3.5 (Visual Studio Community)
Hash: dd394471
Branch: xcode11
Build date: 2019-08-21 07:29:09-0400

=== Xamarin.iOS ===

Version: 12.99.3.5 (Visual Studio Community)
Hash: dd394471
Branch: xcode11
Build date: 2019-08-21 07:29:08-0400

=== Xamarin Designer ===

Version: 16.2.0.414
Hash: a1c119d5f
Branch: remotes/origin/d16-2-xcode11
Build date: 2019-08-20 22:38:28 UTC

=== Xamarin.Android ===

Version: 9.4.1.0 (Visual Studio Community)
Commit: xamarin-android/d16-2/ebd65c9
Android SDK: /Users/ewan/Library/Developer/Xamarin/android-sdk-macosx
	Supported Android versions:
		None installed

SDK Tools Version: 26.1.1
SDK Platform Tools Version: 28.0.2
SDK Build Tools Version: 28.0.3

Build Information: 
Mono: mono/mono/2019-02@e6f5369c2d2
Java.Interop: xamarin/java.interop/d16-2@d64ada5
LibZipSharp: grendello/LibZipSharp/d16-2@caa0c74
LibZip: nih-at/libzip/rel-1-5-1@b95cf3f
ProGuard: xamarin/proguard/master@905836d
SQLite: xamarin/sqlite/3.27.1@8212a2d
Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-2@6f6c969

=== Microsoft Mobile OpenJDK ===

Java SDK: /Users/ewan/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
1.8.0-25
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

=== Android Device Manager ===

Version: 1.2.0.60
Hash: 686c817
Branch: remotes/origin/d16-2
Build date: 2019-08-20 15:19:18 UTC

=== Xamarin Inspector ===

Version: 1.4.3
Hash: db27525
Branch: 1.4-release
Build date: Mon, 09 Jul 2018 21:20:18 GMT
Client compatibility: 1

=== Build Information ===

Release ID: 802040045
Git revision: 7da3e8d5e4b3bf8f0d4cf83f92d60c82094502b2
Build date: 2019-08-21 17:33:58+00
Build branch: release-8.2-xcode11
Xamarin extensions: 12c306582cf9fe4ef451e9b27422c416d11c6491

=== Operating System ===

Mac OS X 10.15.0
Darwin 19.0.0 Darwin Kernel Version 19.0.0
    Fri Aug  9 21:59:46 PDT 2019
    root:xnu-6153.0.139.161.2~2/RELEASE_X86_64 x86_64
```

### Build Logs

The following error is repeated for every single project in the solution (We have about 15) as well as the final application.

```
 AOT Compilation exited with code 139, command:
    MONO_PATH=/Users/ewan/Documents/<path...to...app>/obj/iPhone/Debug/device-builds/iphone8.1-12.4/mtouch-cache/3-Build /Library/Frameworks/Xamarin.iOS.framework/Versions/12.99.3.5/bin/arm64-darwin-mono-sgen --debug -O=gsharedvt -O=-float32  --aot=mtriple=arm64-ios,data-outfile=/Users/ewan/<path...to...app>/obj/iPhone/Debug/device-builds/iphone8.1-12.4/mtouch-cache/arm64/<linked_proejct>.dll.aotdata.arm64,static,asmonly,direct-icalls,full,soft-debug,dwarfdebug,no-direct-calls,outfile=/Users/ewan/Documents/<path...to...app>/obj/iPhone/Debug/device-builds/iphone8.1-12.4/mtouch-cache/arm64/<linked_proejct>.dll.s "/Users/ewan/Documents/<path...to...app>/obj/iPhone/Debug/device-builds/iphone8.1-12.4/mtouch-cache/3-Build/<linked_proejct>.dll"
```

### Example Project (If Possible)

@rolfbjarne has access to our company's repo as a result #6740. Is reproducable with both iOS applications found in there.

**Comment by projectgoav** [comment]
As with my previous experience with my issue #6740, I'll do a complete re-install of Visual Studio and XCode to check my machine is not foobarred.

This was previously working before the most recent XCode 11 Preview channel update I installed this morning.

**Comment by rolfbjarne** [comment]
There are two issues here:

* The projects build for armv7, and we won't support building for armv7 on Catalina for a little while (it won't take long after the stable Xcode 11 / iOS 13 release though, probably just a week or two).
* A variation of #6841, just that the crashing executable is the AOT compiler and not mlaunch.

As such, I'm closing this as a duplicate of #6841, when that bug is fixed, this one will be as well.

**Comment by projectgoav** [comment]
> •	The projects build for armv7, and we won't support building for armv7 on Catalina for a little while (it won't take long after the stable Xcode 11 / iOS 13 release though, probably just a week or two).

I'm a little confused about this point. I've been able to deploy and debug on devices until this morning's Xamarin update.  Does this mean if I want to test on devices going forward I should stay on `12.99.2.40` until XCode 11 preview is available on the stable channel?

**Comment by rolfbjarne** [comment]
> Does this mean if I want to test on devices going forward I should stay on `12.99.2.40` until XCode 11 preview is available on the stable channel?

No, just select "ARM64" in the project's iOS Build options as the architecture (and you'll also have to wait for the fix for #6841). Another alternative is to use Mojave instead of Catalina.

**Comment by projectgoav** [comment]
ok, great. Thanks for the clarification. I'll stick with `12.99.2.40` until #6841 is fixed then :)
